File: backend/src/main/java/com/ironlogic/modules/program/domain/model/SessionExerciseTemplate.java
package com.ironlogic.modules.program.domain.model;

import java.math.BigDecimal;
import java.time.LocalDateTime;

/**
 * SessionExerciseTemplate 的领域模型。
 *
 * <p>这个对象把 SessionTemplate 和 Exercise 关联起来，并保存该位置上的模板处方，
 * 例如目标组数、次数、重量和休息时间。
 *
 * @param id 模板动作 id
 * @param sessionTemplateId 所属 SessionTemplate id
 * @param exerciseId 引用的 Exercise id
 * @param orderNo 在 SessionTemplate 内的展示顺序
 * @param targetSets 计划组数
 * @param targetReps 可选目标次数
 * @param targetWeight 可选目标重量
 * @param targetWeightUnit 可选重量单位
 * @param restSeconds 可选休息秒数
 * @param intensityMode 可选强度模式
 * @param progressionRuleId 预留给未来 progression rule 的 id
 * @param prescriptionJson 预留处方 JSON
 * @param createdAt 创建时间
 * @param updatedAt 更新时间
 */
public record SessionExerciseTemplate(
        Long id,
        Long sessionTemplateId,
        Long exerciseId,
        Integer orderNo,
        Integer targetSets,
        Integer targetReps,
        BigDecimal targetWeight,
        String targetWeightUnit,
        Integer restSeconds,
        String intensityMode,
        Long progressionRuleId,
        String prescriptionJson,
        LocalDateTime createdAt,
        LocalDateTime updatedAt
) {
}

File: frontend/lib/features/program/domain/model/session_exercise_template.dart
/// SessionExerciseTemplate 领域模型。
///
/// 该类属于 `program/domain`，表示训练模板中的一个动作处方。
/// 它仍然是模板定义，而不是某次实际训练的组记录。
class SessionExerciseTemplate {
  /// 创建 SessionExerciseTemplate 模型。
  const SessionExerciseTemplate({
    required this.id,
    required this.sessionTemplateId,
    required this.exerciseId,
    required this.orderNo,
    required this.targetSets,
    this.targetReps,
    this.targetWeight,
    this.targetWeightUnit,
    this.restSeconds,
    this.intensityMode,
    this.progressionRuleId,
    this.prescriptionJson,
    this.createdAt,
    this.updatedAt,
  });

  /// 模板动作 id。
  final int id;

  /// 所属 SessionTemplate id。
  final int sessionTemplateId;

  /// 引用的 Exercise id。
  final int exerciseId;

  /// 顺序号。
  final int orderNo;

  /// 目标组数。
  final int targetSets;

  /// 目标次数。
  final int? targetReps;

  /// 目标重量。
  final double? targetWeight;

  /// 重量单位。
  final String? targetWeightUnit;

  /// 休息秒数。
  final int? restSeconds;

  /// 强度模式。
  final String? intensityMode;

  /// 进阶规则 id。
  final int? progressionRuleId;

  /// 预留处方 JSON。
  final String? prescriptionJson;

  /// 创建时间。
  final DateTime? createdAt;

  /// 更新时间。
  final DateTime? updatedAt;
}
